test/poly/spoly-test.jl

```diff
@@ -454,6 +454,16 @@ function test_spoly_differential()
    @test jf == x^3
 end
 
+function test_spoly_unique()
+   print("spoly.unique...")
+   
+   R, (x, y) = PolynomialRing(QQ, ["x", "y"])
+
+   @test unique([x, x+y-y]) == [x]
```

Comment:
this test will occasionally pass even without overloaded hash function; 
we should add test which directly compares hashes
in this case these should be the same (and we don't have negative example due to trivial definition above)

Reply:
See 1bb108f.

Reply:
So is `test_spoly_unique` a useful test at all? It only tests one particular symptom, not the core issue, which is tested in `test_spoly_hash`. Is there another reason to test `unique` on polynomials like this?